src/wasm/wat-parser.cpp

```diff
@@ -421,6 +474,205 @@ template<typename Ctx>
 MaybeResult<typename Ctx::StructT> structtype(Ctx&, ParseInput&);
 template<typename Ctx>
 MaybeResult<typename Ctx::ArrayT> arraytype(Ctx&, ParseInput&);
+template<typename Ctx>
+Result<typename Ctx::GlobalTypeT> globaltype(Ctx&, ParseInput&);
+
+// Instructions
+template<typename Ctx>
```

Comment:
These declarations could be autogenerated using an include of `wasm-delegations.def`

Reply:
This would be tricky because some of them have extra arguments, such as the calls with their `isReturn` arguments, i31 and struct get operations with their `signed_` argument, and various other instructions with operand arguments.

Reply:
Could we get the `isReturn` from the name, if we read it again? Or is it lost at that point?

Reply:
(In the old parser that was possible, though not sure if we actually used that ability or not)